iroh-util/src/lock.rs

```diff
@@ -28,7 +31,7 @@ impl ProgramLock {
 
         // Even if we manage to lock the file this won't last since the drop implementation
         // of FileGuard releases the underlying lock.
-        if let Ok(file) = File::create(&self.path) {
+        if let Ok(file) = File::open(&self.path) {
```

Comment:
this was creating a bug where calling `is_locked` was overwriting the lock itself. I'm unclear on if that would actually break the lock, but it certainly overwrites the data within the file

Reply:
Should use https://doc.rust-lang.org/std/fs/struct.OpenOptions.html to specify to only create if not exists probably